make/jdk/src/classes/build/tools/generatelsrequivmaps/EquivMapsGenerator.java

```diff
@@ -53,21 +54,31 @@
 public class EquivMapsGenerator {
 
     public static void main(String[] args) throws Exception {
-        if (args.length != 3) {
+        int i = 0;
+        if (args.length == 5 && "-jdk-header-template".equals(args[i])) {
+            jdkHeaderTemplate = new String(Files.readAllBytes(Paths.get(args[++i])),
+                    StandardCharsets.UTF_8);
+            i++;
+        } else if (args.length != 3) {
```

Comment:
This is not very robust for detecting malformed invocations. It won't even detect a simple typo in the argument name. I would suggest something more elaborate e.g.
```
int i  0;
boolean valid = true;
if (args.length != 5 && args.length !=3) {
  valid = false;
} else if (args.length == 5) {
  if ( ! "-jdk-header-template".equals(args[i])) {
    valid = false;
  } else {
    jdkHeaderTemplate = new String(Files.readAllBytes(Paths.get(args[++i])),
                    StandardCharsets.UTF_8);
     i++;
  }
}
if (!valid) {
  System.err.println(...);
   System.exit(1);
}
String lsrFile = args[i++];
...
```

Reply:
I think it actually will, but your suggestion is clearer to read, so I have applied it.